### PR TITLE
Update checkbashism-native to 2.23.6

### LINKED ISSUE #11714: Update checkbashism-native to 2.23.6

### PR DESCRIPTION
**Pull request by kk6ho**
Closes #11714

### CODE REVIEW

**Comment by kk6ho** [comment]
This is a cherry pick of Konrad's commit 030843aabfe6feee17b3f013a82930afdf6e7b3a from the master branch  to the kirkstone branch.

Fixes fetch error when using the meta-sca kirkstone branch due to devscripts_2.23.5.tar.xz file no longer at the SRC_URI site (versions 2.23.4 and 2.23.6 are on that site).

File: recipes-sca/checkbashism-native/nativesdk-checkbashism_2.23.6.bb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+SUMMARY = "check for bashisms in /bin/sh scripts"
```
Comment (priv-kweihmann) [inline recipes-sca/checkbashism-native/nativesdk-checkbashism_2.23.6.bb:1]: IIRC there's no support for nativesdk tooling in kirkstone, so I guess it's better to drop this file here

**Review by priv-kweihmann — CHANGES_REQUESTED** [review]

File: recipes-sca/checkbashism-native/nativesdk-checkbashism_2.23.6.bb (hunk 1 above)
Comment (kk6ho) [reply recipes-sca/checkbashism-native/nativesdk-checkbashism_2.23.6.bb:1]: When I cherry-picked from master I did get a merge conflict due to the nativesdk-checkbashism_2.23.6.bb not previously existing, so I accepted adding it as an new file.  I can remove this file from the kirkstone commit if that is your preference?

Comment (kk6ho) [reply recipes-sca/checkbashism-native/nativesdk-checkbashism_2.23.6.bb]: I removed  the new file **nativesdk**-checkbashism_2.33.6.bb from the PR

**Review by priv-kweihmann — APPROVED** [review]

**Pull request merged**